src/hooks/useLoginProfile.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  CodeChallengeMethod,
+  exchangeCodeAsync,
+  makeRedirectUri,
+  refreshAsync,
+  ResponseType,
+  useAuthRequest
+} from 'expo-auth-session';
+import { useEffect, useState } from 'react';
+
+import * as appJson from '../../app.json';
+import { secrets } from '../config';
+import { addToStore, readFromStore } from '../helpers';
+
+const PROFILE_ACCESS_TOKEN = 'profileAccessToken';
+
+const namespace = appJson.expo.slug as keyof typeof secrets;
+const profile = secrets[namespace]?.profile;
+const serverUrl = profile?.serverUrl;
+
+const keycloakDiscovery = {
+  authorizationEndpoint: serverUrl + '/auth',
+  tokenEndpoint: serverUrl + '/token',
+  revocationEndpoint: serverUrl + '/revoke'
+};
+
+const CLIENT_ID = profile?.clientId;
+const CLIENT_SECRET = profile?.clientSecret;
+const SCOPES = ['openid', 'email', 'profile', 'phone', 'address'];
+
+export const useLoginProfile = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const redirectUri = makeRedirectUri({
+    scheme: appJson.expo.scheme,
+    path: 'redirect'
+  });
+
+  const [request, response, promptAsync] = useAuthRequest(
+    {
+      clientId: CLIENT_ID,
+      codeChallengeMethod: CodeChallengeMethod.S256,
+      redirectUri,
+      responseType: ResponseType.Code,
+      scopes: SCOPES,
+      usePKCE: false
```

Comment:
why do we set this to `false`?

Reply:
where do you know that from?

Reply:
Because the pkce value is off in keycloak. But I don't know why it's off.

<img width="312" alt="image" src="https://github.com/user-attachments/assets/af53c344-8d1d-442a-aa83-86fc967e29ee" />

You can see this setting by following the route below.
keycloak -> Identity providers -> ITEBO

Reply:
https://oauth.net/2/pkce, `true` per default: https://docs.expo.dev/versions/latest/sdk/auth-session/#authrequestconfig

so if someone changes this in keycloak, we need a new app release with also an adjusted value. can we consider this to be part of the global settings?

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1311/commits/ddabdcba94f62db4dbb1ad567f24403cc2e39ef8